datanode/sqlstore/orders.go

```diff
@@ -98,6 +99,39 @@ func (os *Orders) GetOrder(ctx context.Context, orderIDStr string, version *int3
 	return order, os.wrapE(err)
 }
 
+// GetByMarketAndID returns all orders with given IDs for a market.
+func (os *Orders) GetByMarketAndID(ctx context.Context, marketIDstr string, orderIDs []string) ([]entities.Order, error) {
+	if len(orderIDs) == 0 {
+		os.log.Warn("GetByMarketAndID called with an empty order slice",
+			logging.String("market ID", marketIDstr),
+		)
+		return nil, nil
+	}
+	defer metrics.StartSQLQuery("Orders", "GetByMarketAndID")()
+	marketID := entities.MarketID(marketIDstr)
+	// IDs := make([]entities.OrderID, 0, len(orderIDs))
+	IDs := make([]interface{}, 0, len(orderIDs))
+	in := make([]string, 0, len(orderIDs))
+	bindNum := 2
+	for _, o := range orderIDs {
+		IDs = append(IDs, entities.OrderID(o))
+		in = append(in, fmt.Sprintf("$%d", bindNum))
+		bindNum++
+	}
+	bind := make([]interface{}, 0, len(in)+1)
+	// set all bind vars
+	bind = append(bind, marketID)
+	bind = append(bind, IDs...)
```

Comment:
Have you seen the `nextBindVar` helper function, it makes keeping track of these bind numbers much simpler

```go
func (s *Store) DoQuery(thing, wotsit string) {
	args := []interface{}{}
	query := fmt.Sprintf("select * from foo where thing=%s and whotsit=%s",
		nextBindVar(&args, thing),
		nextBindVar(&args, wotsit)
	)
}
```